tests/h/traversal/annotation_test.py

```diff
@@ -1,348 +1,134 @@
 from pyramid import security
 from pyramid.authorization import ACLAuthorizationPolicy
 
 from h.security.permissions import Permission
 from h.traversal import AnnotationContext, AnnotationRoot
 
 
-class FakeGroup:
-    # NB: Tests that use this do not validate that the principals are correct
-    # for the indicated group. They validate that those principals are being
-    # transferred over to the annotation as expected
-    # As such, this has sort of a partial and wonky re-implementation of
-    # ``h.models.Group.__acl__``
-    # This is a symptom of the disease that is splitting ACL concerns between
-    # traversal/resources and model classes
-    # TODO: Refactor once we're able to move ACLs off of models
-    def __init__(self, principals):
-        acl = []
-        for p in principals:
-            acl.append((security.Allow, p, Permission.Group.READ))
-            if p == security.Everyone:
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.FLAG)
-                )
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.MODERATE)
-                )
-            else:
-                acl.append((security.Allow, p, Permission.Group.FLAG))
-                # Normally, the ``moderate`` permission would only be applied
-                # to the admin (creator) of a group, but this ``FakeGroup``
-                # is indeed fake. Tests in this module are merely around whether
-                # this permission is translated appropriately from a group
-                # to an annotation context (i.e. it should not be applied
-                # to private annotations)
-                acl.append((security.Allow, p, Permission.Group.MODERATE))
-        self.__acl__ = acl
-
-
-@pytest.mark.usefixtures("groupfinder_service", "links_service")
 class TestAnnotationRoot:
-    def test_it_does_not_assign_create_permission_without_authenticated_user(
-        self, set_permissions, pyramid_request
-    ):
-        set_permissions()
-
-        context = AnnotationRoot(pyramid_request)
+    def test_create_permission_requires_authenticated_user(self, root, permits):
+        assert permits(root, [security.Authenticated], Permission.Annotation.CREATE)
+        assert not permits(root, [], Permission.Annotation.CREATE)
 
-        assert not pyramid_request.has_permission(Permission.Annotation.CREATE, context)
-
-    def test_it_assigns_create_permission_to_authenticated_request(
-        self, set_permissions, pyramid_request
+    def test_annotation_lookup(
+        self,
+        root,
+        pyramid_request,
+        AnnotationContext,
+        storage,
+        groupfinder_service,
+        links_service,
     ):
-        set_permissions(
-            "acct:adminuser@foo", principals=[pyramid.security.Authenticated]
+        context = root[sentinel.annotation_id]
+
+        assert context == AnnotationContext.return_value
```

Comment:
I'd normally put this one at the bottom so that the assertions are in the order that the things happen in the code: first it calls `fetch_annotation()`, then it calls `AnnotationContext()` (passing it the result of `fetch_annotation()`), then it returns the context. I just think that assertion order makes the test easier to comprehend

Reply:
Yeah, I was working backwards from some existing tests and merging them together in a hurry so they are a bit of a mess. I've also been looking at these all so long I'm going crosseyed.